Examples/Python/python/acts/examples/__init__.py

```diff
@@ -441,21 +472,164 @@ def _getAutoFpeMasks(cls) -> List[FpeMask]:
-                                        count,
-                                    )
+                    lines = fh.readlines()
+                for i, line in enumerate(lines):
+                    if m := re.match(r".*\/\/ ?MARK: ?(fpeMask\(.*)$", line):
+                        exp = m.group(1)
+                        for m in re.findall(
+                            r"fpeMask\( ?(\w+), ?(\d+) ?, ?#(\d+) ?\)", exp
+                        ):
+                            fpeType, count, _ = m
+                            count = int(count)
+                            rel = f.relative_to(srcdir)
+                            cls._autoFpeMasks.append(
+                                cls.FpeMask(
+                                    str(rel),
+                                    (i + 1, i + 2),
+                                    cls.FpeMask._fpe_types_to_enum[fpeType],
+                                    count,
                                 )
+                            )
+
+                    if m := re.match(
+                        r".*\/\/ ?MARK: ?fpeMaskBegin\( ?(\w+), ?(\d+) ?, ?#?(\d+) ?\)",
+                        line,
+                    ):
+                        fpeType, count, _ = m.groups()
+                        count = int(count)
+                        rel = f.relative_to(srcdir)
+
+                        start = i + 1
+                        end = None
+
+                        # look for end marker
+                        for j, line2 in enumerate(lines[i:]):
+                            if m := re.match(
+                                r".*\/\/ ?MARK: ?fpeMaskEnd\( ?(\w+) ?\)$", line2
+                            ):
+                                endType = m.group(1)
+                                if endType == fpeType:
+                                    end = i + j + 1
+                                    break
+
+                        if end is None:
+                            raise ValueError(
+                                f"Found fpeMaskBegin but no fpeMaskEnd for {rel}:{start}"
+                            )
+                        cls._autoFpeMasks.append(
+                            cls.FpeMask(
+                                str(rel),
+                                (start, end + 1),
+                                cls.FpeMask._fpe_types_to_enum[fpeType],
+                                count,
+                            )
+                        )
 
         return cls._autoFpeMasks
+
+    @classmethod
+    def _printFpeSummary(cls, masks: List[FpeMask]):
+        if len(masks) == 0:
+            return
+
+        # Try to make a nice summary with rich, or fallback to a plain text one
+        try:
+            import rich
+
+            have_rich = True
+        except ImportError:
+            have_rich = False
+
+        error = False
+        srcdir = cls.srcdir()
+
+        def rich_summary():
```

Comment:
since this is only called once and would be intended at the same level I would maybe inline it but not sure

Reply:
Can do.